### ISSUE TITLE
issue: warning "Cannot invoke 'uname` to get architecture type" emitted

### ISSUE THREAD

**Issue by MarioninD** [revoltchat/self-hosted#104]
### What happened?

I can't quite understand what the error is related to, I think I did everything according to the instructions :)

docker-compose logs api and minio/mc
```
Attaching to revolt_api_1
api_1            |  INFO  revolt_config > Starting revolt-delta@0.7.16
api_1            |  WARN  revolt_config > No Captcha key specified! Remember to add hCaptcha key.
api_1            |  **ERROR os_info::architecture > Cannot invoke 'uname` to get architecture type: Os { code: 2, kind: NotFound, message: "No such file or directory" }**
api_1            |  INFO  revolt_database::models::admin_migrations::ops::mongodb > Migrating the database.



createbuckets_1  | mc: <ERROR> Couldn't construct anonymous client for `minio`. No valid configuration found for 'minio' host alias.
createbuckets_1  | Waiting minio...
```

**Comment by JoCat** [comment]
Check if you can invoke the `uname` command on your system

**Comment by Sirse** [comment]
Same issue. uname and uname -a works fine in host machine

**Comment by Deswolrd** [comment]
Not sure if it's the same issue, but looks quite the same for me. Happens when trying to connect WebSocket

```
events-1         |  INFO  revolt_config > Starting revolt-bonfire@0.7.16
events-1         |  ERROR os_info::architecture > Cannot invoke 'uname` to get architecture type: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

Right after that, main thread collapses
```
events-1         | thread 'main' panicked at /home/rust/src/crates/core/presence/src/lib.rs:166:43:
events-1         | Redis connection: FailedConnection
events-1         | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

uname in host machine looks good as far as I can tell
```
uname -a
Linux debian 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux
```

Fresh setup with .env / Revolt.toml changes for external access via reverse proxy (curl and wscat from local machine directly provides the same results). Used mongo version (4.4 for older CPU as guide says).

**Comment by Sirse** [comment]
> Not sure if it's the same issue, but looks quite the same for me. Happens when trying to connect WebSocket
> 
> ```
> events-1         |  INFO  revolt_config > Starting revolt-bonfire@0.7.16
> events-1         |  ERROR os_info::architecture > Cannot invoke 'uname` to get architecture type: Os { code: 2, kind: NotFound, message: "No such file or directory" }
> ```
> 
> Right after that, main thread collapses
> 
> ```
> events-1         | thread 'main' panicked at /home/rust/src/crates/core/presence/src/lib.rs:166:43:
> events-1         | Redis connection: FailedConnection
> events-1         | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> uname in host machine looks good as far as I can tell
> 
> ```
> uname -a
> Linux debian 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux
> ```
> 
> Fresh setup with .env / Revolt.toml changes for external access via reverse proxy (curl and wscat from local machine directly provides the same results). Used mongo version (4.4 for older CPU as guide says).

Yes, absolutely same issue (Ubuntu 24.04.1 LTS), old Intel NUC with mongo 4.4.

**Comment by JVT038** [comment]
Same error here, on Synology DS918+.

**Comment by Macarthurval** [comment]
Same issue here.

```
uname -a
Linux revolt-chat 5.14.0-513.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Sep 27 07:50:59 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by brownian-motion** [comment]
I'm hosting on Linux/Ubuntu and ran into the same problem.

The error message is created in the `os_info` library, when i tries to run `uname`: https://github.com/stanislav-tkach/os_info/blob/master/os_info/src/architecture.rs#L10
Because the `revolt` image is stripped-down and has no executables, this library call fails to execute `uname` and fails.

I was able to work around it by modifying my `compose.yml` to manually copy `uname` from the original image used to build `revolt` in the first place.

Specifically, I made this change to my `compose.yml`:

BEFORE
```yml
  # API server (delta)
  api:
    image: ghcr.io/revoltchat/server:20240929-1
```

```
  # API server (delta)
  api:
    build:
      dockerfile_inline: |
        FROM rust:1.77.2-slim-bookworm AS builder
        FROM ghcr.io/revoltchat/server:20240929-1 AS target
        COPY --from=builder /bin/uname /usr/bin/uname
```

**Comment by IAmTomahawkx** [comment]
This is fixed in build 20241212-1
